### PR TITLE
Introudce State migration periodic log

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR introduce periodic log in state migration.
Batch write while migration occures when the size reaches `IdealBatchSize`.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/state_migration.go [hunk 1]
```diff
@@ -52,20 +53,20 @@ func (td *stateTrieMigrationDB) ReadPreimage(hash common.Hash) []byte {
...
-func (bc *BlockChain) stateMigrationCommit(s *statedb.TrieSync, db database.DBManager) (int, time.Duration, error) {
-	start := time.Now()
-	stateTrieBatch := db.NewBatch(database.StateTrieDB)
-
-	written, err := s.Commit(stateTrieBatch)
+func (bc *BlockChain) stateMigrationCommit(s *statedb.TrieSync, batch database.Batch) (int, error) {
+	written, err := s.Commit(batch)
 	if written == 0 || err != nil {
-		return written, 0, err
+		return written, err
```
Comment (aidan-kwon) [inline blockchain/state_migration.go:59]: Can we sure that `written` is always bigger than 0 in normal cases?

File: blockchain/state_migration.go [hunk 2]
```diff
@@ -183,27 +184,76 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
...
+		return fmt.Errorf("DB write error: %v", err)
+	}
+
+	stats.stateMigrationReport(true, trieSync.Pending(), trieSync.CalcProgressPercentage())
 
 	elapsed := time.Since(start)
-	speed := float64(committedCnt) / elapsed.Seconds()
-	logger.Info("State migration is completed", "committedCnt", committedCnt, "elapsed", elapsed, "committed per second", speed)
+	speed := float64(stats.totalCommitted) / elapsed.Seconds()
+	logger.Info("State migration : copy is done",
```
Comment (aidan-kwon) [inline blockchain/state_migration.go:197]:
```suggestion
	logger.Info("State migration : Copy is done",
```

File: blockchain/state_migration.go [hunk 3]
```diff
@@ -144,32 +148,29 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
...
-		bc.committedCnt, bc.pendingCnt, bc.progress = committedCnt, trieSync.Pending(), trieSync.CalcProgressPercentage()
-		progressStr := strconv.FormatFloat(bc.progress, 'f', 4, 64)
-		progressStr = strings.TrimRight(progressStr, "0")
-		progressStr = strings.TrimRight(progressStr, ".") + "%"
-
-		logger.Warn("State migration progress",
-			"progress", progressStr, "committedCnt", committedCnt, "pendingCnt", bc.pendingCnt,
-			"read", read, "readElapsed", readElapsed, "written", written, "writeElapsed", writeElapsed,
-			"elapsed", time.Since(startIter))
+		stats.stateMigrationReport(false, trieSync.Pending(), trieSync.CalcProgressPercentage())
```
Comment (aidan-kwon) [inline blockchain/state_migration.go:173]: We need to change `CalcProgressPercentage()` if the size of `trieSync.Missing` is changed.

File: blockchain/state_migration.go [hunk 4]
```diff
@@ -122,14 +123,17 @@ func (bc *BlockChain) migrateState(rootHash common.Hash) error {
...
 	}
 
+	stateTrieBatch := targetDB.DiskDB().NewBatch(database.StateTrieDB)
+	stats := migrationStats{initialStartTime: start, startTime: mclock.Now()}
+
 	// Migration main loop
 	for trieSync.Pending() > 0 {
 		bc.committedCnt, bc.pendingCnt = committedCnt, trieSync.Pending()
-		queue = append(queue[:0], trieSync.Missing(database.IdealBatchSize)...)
+		queue = append(queue[:0], trieSync.Missing(1024)...)
```
Comment (aidan-kwon) [inline blockchain/state_migration.go:132]: We need to change CalcProgressPercentage() if the size of trieSync.Missing is changed.
I think the accuracy of depth 3 is enough.

File: blockchain/state_migration.go (hunk 1 above)
Comment (KimKyungup) [reply blockchain/state_migration.go:59]: Yes, it is.
`written` is `written := len(s.membatch.order)`.
So it is always 0 or more.

If the all result has children, 0 `written` is possible.

File: blockchain/state_migration.go (hunk 4 above)
Comment (KimKyungup) [reply blockchain/state_migration.go:132]: Good. I reflected.

**Comment by KimKyungup** [comment]
PTAL :)

**Comment by KimKyungup** [comment]
PTAL. :)

**Pull request merged**